rules.py

```diff
@@ -410,12 +409,14 @@ def check(self, element):
             return
         valid_ocd_id = False
         referenced_gpunit = None
+        external_ids = []
         for gpunit in self.gpunits:
             if gpunit.get("objectId", None) == element.text:
                 referenced_gpunit = gpunit
-                for extern_id in gpunit.iter("ExternalIdentifier"):
+                external_ids = gpunit.findall(".//ExternalIdentifier")
+                for extern_id in external_ids:
                     id_type = extern_id.find("Type")
-                    if id_type is not None and id_type.text == "ocd-id":
+                    if id_type is not None and id_type.text.lower() == "ocd-id":
```

Comment:
I'm concerned about this because I'm pretty sure that if a publisher uses <Type>OCD-ID</Type> we won't be able to ingest it because our Java pipeline will reject it as being invalid. Perhaps include a separate check:

if id_type.text.lower() == "ocd-id" and id_type.text.lower() != id_type.text:
  // Error message about improperly-cased ExternalIdentifier Type